plugins/csharp/Managed/VideoSource.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Neurala {
+    public static class NativeVideoSource {
+        [StructLayout(LayoutKind.Sequential)]
+        private struct StructureForGetMetadata {
+            public int Width;
+            public int Height;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct StructureForMoveNextFrame {
+            public int Status;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct StructureForGetFrame {
+            public IntPtr Data;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct StructureForExecute {
+            public IntPtr Action;
+        }
+
+        public static void GetMetadata(IntPtr result, int size) {
+            var (width, height) = GetMetadata();
+
+            var structure = new StructureForGetMetadata { Width = width, Height = height };
+
+            Marshal.StructureToPtr<StructureForGetMetadata>(structure, result, false);
+        }

```

Comment:
In practice, `Bitmap` images in .NET are almost always plain old RGB or RGBA, so I'm wondering whether just dimension information would be enough.